src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,201 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> head;
+    private Node<T> tail;
+    private int size = 0;
+
     @Override
     public void add(T value) {
+        if (size >= 1) {
+            Node<T> prevTale = tail;
+            tail = new Node<>(prevTale, value, null);
+            prevTale.next = tail;
+        } else {
+            Node<T> fistNode = new Node<>(null, value, null);
+            head = fistNode;
+            tail = fistNode;
+        }
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
```

Comment:
method is overengineered, can be done with one if-else

Reply:
This method is already done with one if-else, if I understand you correctly.